org/mozilla/jss/ssl/SSLAlertEvent.java

```diff
@@ -98,6 +100,20 @@ public void setDescription(SSLAlertDescription description) {
         this.description = description.getID();
     }
 
+    public void throwException() throws SSLException {
+        if (levelEnum == SSLAlertLevel.FATAL) {
+            Class exception = descriptionEnum.getExceptionClass();
+
+            try {
+                throw (SSLException) exception.getConstructor(String.class).newInstance(this.toString());
+            } catch (Exception e) {
```

Comment:
I suppose this will catch the `SSLException` itself, so you might want to create the instance in the `try-catch` block, but throw it outside the block. The `catch` block should catch the exception thrown by `getConstructor()` and `newInstance()` and chain it to a new `SSLException`.

Reply:
Ah, yeah, good catch.